bdd-for-c.h

```diff
@@ -327,23 +271,93 @@ typedef struct __bdd_config_type__ {
 
 char *__bdd_spec_name__;
 void __bdd_test_main__(__bdd_config_type__ *__bdd_config__);
+char *__bdd_vformat__(const char *format, va_list va);
+
+void __bdd_indent__(FILE *fp, size_t level) {
+    for (size_t i = 0; i < level; ++i) {
+        fprintf(fp, "  ");
+    }
+}
+
+bool __bdd_enter_node__(__bdd_config_type__ *config, __bdd_node_type__ type, ptrdiff_t list_offset, char *fmt, ...) {
+    va_list va;
+    va_start(va, fmt);
+    char *name = __bdd_vformat__(fmt, va);
+    va_end(va);
+
+    if (config->run == __BDD_INIT_RUN__) {
+        __bdd_node__ *top = __bdd_array_last__(config->node_stack);
+        __bdd_array__ *list = *(__bdd_array__ **)((unsigned char *)top + list_offset);
+
+        int id = config->id++;
+        __bdd_node__ *node = __bdd_node_create__(id, name, type);
+        __bdd_array_push__(list, node);
+        __bdd_array_push__(config->nodes, node);
+        if (type == __BDD_NODE_GROUP__) {
+            __bdd_array_push__(config->node_stack, node);
+            return true;
+        }
+        return false;
+    }
+
+    if (config->id >= config->nodes->size) {
```

Comment:
MSVC reports signed / unsigned mismatch here

Reply:
Got the same warnings with gcc too with `-Wall -Wextra -pedantic`. I will use them from now on.

I fixed this warning by casting the right hand side to `int`. `config->id` tracks the sequential IDs of tests and is assumed to be wide enough not to overflow. The size of the nodes array should be less than this and this line is a sanity check that we don't end up in a code path in `__bdd_test_main__` with an unexpected number of tests for some reason.

I decided against making the IDs unsigned or size_t since the root node has ID -1. Its ID doesn't really matter since we never come across it with `__BDD_STEP__`, but it might be confusing if, for instance, there are two nodes with ID 0. It's not part of the nodes array and its allocation is handled separately. However, there was not much thought behind using the `int` type for IDs. Maybe an unsigned type would be better?

There were more files where similar signed/unsigned errors happened (`before-after.c` and `dynamic-test.c`) which were also fixed.